packages/devextreme/js/__internal/core/state_manager/state_manager.ts

```diff
@@ -0,0 +1,199 @@
+          }
+
+          if (typeof newValue === 'object' && newValue !== null) {
+            valueContainerChangeCopy.payload.newValue = deepCopy(newValue);
+          }
+
+          const updatedComponentState = this.getComponentState();
+
+          this.devToolsConnector
+            .sendAction(
+              'UPDATE',
+              valueContainerChangeCopy.payload,
+              updatedComponentState,
+            );
+        },
+      );
+    } catch (error) {
+      this.logger.error(`Failed to track state for ${fullPathToProperty}`, error);
+    }
+  }
+
+  private hasValueContainerManagerFor(
+    valueContainer: StateManagementTypes.MaybeValueContainer,
+  ): boolean {
+    return this.valueContainerManagers
+      // eslint-disable-next-line @stylistic/max-len
+      .some((currentStateContainerManager) => currentStateContainerManager.canHandle(valueContainer));
+  }
+
+  private findValueContainerManagerFor(
+    valueContainer: StateManagementTypes.MaybeValueContainer,
+  ): StateManagementTypes.ValueContainerManager | undefined {
+    const valueContainerManagerFactory = this.valueContainerManagers
+      // eslint-disable-next-line @stylistic/max-len
+      .find((currentStateContainerManager) => currentStateContainerManager.canHandle(valueContainer));
+
+    if (!valueContainerManagerFactory) {
+      return undefined;
+    }
+
+    return valueContainerManagerFactory.create(this.logger, this.stateSourceSign, valueContainer);
+  }
+
+  getComponentState(): StateManagementTypes.ComponentState {
+    const result = Object.entries(this.componentState)
+      .reduce((acc, [stateId, stateValue]) => {
+        Object.entries(stateValue).forEach(([propertyName, propertyValue]) => {
+          const preparedPropertyValue = propertyValue instanceof WeakRef
+            // eslint-disable-next-line spellcheck/spell-checker
+            ? propertyValue.deref() : propertyValue;
+
+          if (!preparedPropertyValue) {
+            return acc;
+          }
+
+          const valueContainerManager = this.findValueContainerManagerFor(preparedPropertyValue);
+
+          if (!valueContainerManager) {
+            return acc;
+          }
+
+          const value = valueContainerManager.getValue();
+
+          if (!acc[stateId]) {
+            acc[stateId] = {};
+          }
+
+          acc[stateId][propertyName] = value !== undefined && value !== null && typeof value === 'object' ? deepCopy(value) : value;
+
+          return acc;
+        });
+
+        return acc;
+      }, {});
+
+    return result;
+  }
+}
+
+export const StateManagerFactory = {
+  create: (options: StateManagementTypes.StateManagerFactoryOptions): StateManager => {
+    const logger = options.logger ?? new Logger({ logLevel: options.logLevel, prefix: '[StateManager]' });
+
+    const stateContainerManagers: StateManagementTypes.StateManagerConfig['valueContainerManagers'] = options.valueContainerManagers ?? [PreactSignalValueContainerManagerFactory];
```

Comment:
Let's import directly desired type (it's exported from `types.ts`)
```suggestion
    const stateContainerManagers: ValueContainerManagerConstructor = options.valueContainerManagers[] ?? [PreactSignalValueContainerManagerFactory];
```

Reply:
This was done in the context of the state manager config. Why is it necessary to go under the hood of public API details?
In the current approach, if I change the type of the API, I don't need to change this line of code.

Reply:
Ok, let's leave it as is